src/plugins/elasticsearch/lib/create_agent.js

```diff
@@ -1,30 +1,28 @@
 var url = require('url');
-var fs = require('fs');
+var _ = require('lodash');
+var readFile = _.partialRight(require('fs').readFileSync, 'utf8');
 var http = require('http');
-var agentOptions;
-module.exports = function (server) {
+var https = require('https');
+
+module.exports = _.memoize(function (server) {
```

Comment:
Memoize doesn't work with objects, does it? Pretty sure it stringifies these. You'll probably need to set `module.exports.cache` to be a `new Map()` or something for this to work right.